pep-9999.rst

```diff
@@ -268,44 +257,51 @@ objects, ignoring missing files and merging the documents into a single object).
 Doing this would be inconsistent with ``json.load``, ``pickle.load``, etc. If we
 agree consistency with other stdlib modules is desirable, allowing paths is
 somewhat out of scope for this PEP. This can easily and explicitly be worked
-around in user code.
+around in user code, or a third-party library.
 
-The proposed API takes a ``SupportsRead[bytes]``, while ``toml.load`` takes a
-``SupportsRead[str]`` and ``json.load`` takes ``SupportsRead[str | bytes]``.
-Using ``SupportsRead[bytes]`` allows us to a) ensure utf-8 is the encoding used,
+The proposed API takes a binary file, while ``toml.load`` takes a
+text file and ``json.load`` takes either.
+Using a binary file allows us to a) ensure utf-8 is the encoding used,
 b) avoid incorrectly parsing single carriage returns as valid TOML due to
 universal newlines.
 
+TODO: why does ``tomllib.loads`` take str but not bytes, then??
```

Comment:
The way I see it, `load` loads a TOML file. A TOML file is strictly UTF-8 and incompatible with the transformations made by the Universal Newlines feature, so it makes no sense for the API to give users the freedom to fiddle with a text file object's options.

A TOML file is, quoting TOML spec
> A TOML file must be a valid UTF-8 encoded Unicode document.

My thinking is that `loads` is not used to load a TOML file, but rather the document that the file stores. The most natural representation for a Unicode document in Python is `str` not `bytes`.

I'm not fully opposed to `loads` being able to load a TOML file rather than the document (i.e. adding `bytes` support) but nobody asked for this feature.

Reply:
```suggestion
Type accepted by the first argument of ``tomllib.loads``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

While ``tomllib.load`` takes a binary file, ``tomllib.loads`` takes
a text string. This may seem inconsistent at first.

A TOML file is, quoting TOML spec

> A TOML file must be a valid UTF-8 encoded Unicode document.

The `loads` function is not used to load a TOML file, but rather the
document that the file stores. The most natural representation for
a Unicode document in Python is `str`, not `bytes`.

It is possible to add `bytes` support in the future if needed, but
are not aware of any use cases for it.
```

Reply:
Great! :+1: I made some minor improvements.
```suggestion
Type accepted by the first argument of ``tomllib.loads``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

While ``tomllib.load`` takes a binary file, ``tomllib.loads`` takes
a text string. This may seem inconsistent at first.

Quoting TOML v1.0.0 specification:

> A TOML file must be a valid UTF-8 encoded Unicode document.

``tomllib.loads`` does not intend to load a TOML file, but rather the
document that the file stores. The most natural representation of
a Unicode document in Python is ``str``, not ``bytes``.

It is possible to add ``bytes`` support in the future if needed, but
we are not aware of any use cases for it.
```